runhouse/rns/hardware/on_demand_cluster.py

```diff
@@ -382,3 +382,83 @@ def ssh_creds(self):
     def ssh(self):
         """SSH into the cluster."""
         subprocess.run(["ssh", f"{self.name}"])
+
+
+# OnDemandCluster factory method
+def autocluster(
+    name: str,
+    instance_type: Optional[str] = None,
+    num_instances: Optional[int] = None,
+    provider: Optional[str] = None,
+    autostop_mins: Optional[int] = None,
+    use_spot: bool = False,
+    image_id: Optional[str] = None,
+    region: Optional[str] = None,
+    dryrun: bool = False,
+) -> OnDemandCluster:
+    """
+    Builds an instance of :class:`OnDemandCluster`.
+
+    Args:
+        name (str): Name for the cluster, to re-use later on.
+        instance_type (int, optional): Type of cloud instance to use for the cluster. This could
+            be a Runhouse built-in type, or your choice of instance type.
+        num_instances (int, optional): Number of instances to use for the cluster.
+        provider (str, optional): Cloud provider to use for the cluster.
+        autostop_mins (int, optional): Number of minutes to keep the cluster up after inactivity,
+            or ``-1`` to keep cluster up indefinitely.
+        use_spot (bool, optional): Whether or not to use spot instance.
+        image_id (str, optional): Custom image ID for the cluster.
+        region (str, optional): The region to use for the cluster.
+        dryrun (bool): Whether to create the Cluster if it doesn't exist, or load a Cluster object as a dryrun.
+            (Default: ``False``)
+
+    Returns:
+        OnDemandCluster: The resulting cluster.
+
+    Example:
+        >>> # On-Demand SkyPilot Cluster (OnDemandCluster)
+        >>> gpu = rh.autocluster(name='rh-4-a100s',
+        >>>                  instance_type='A100:4',
+        >>>                  provider='gcp',
+        >>>                  autostop_mins=-1,
+        >>>                  use_spot=True,
+        >>>                  image_id='my_ami_string',
+        >>>                  region='us-east-1',
+        >>>                  )
+    """
+    if (
+        instance_type is None
```

Comment:
can be simplified with something like 

```
if (
    set(instance_type, num_instances, provider, autostop_mins, image_id, region) == {None}
    and not use_spot and dryrun
)
```

or alternatively `all(param is None for param in (instance_type, num_instances, provider, autostop_mins, image_id, region))` is more intuitive but a little bit slower